src/Cake.Common/Build/AzurePipelines/Data/AzurePipelinesPullRequestInfo.cs

```diff
@@ -41,7 +41,7 @@ public AzurePipelinesPullRequestInfo(ICakeEnvironment environment)
         /// <value>
         ///   The ID of the pull request that caused this build.
         /// </value>
-        public int Id => GetEnvironmentInteger("SYSTEM_PULLREQUEST_PULLREQUESTID");
+        public long Id => GetEnvironmentLongInteger("SYSTEM_PULLREQUEST_PULLREQUESTID");
```

Comment:
We're not keen on a breaking change and a major new version, so non-breaking suggestion  is to add a new property, obsolete the current with message and no error, and we'll take the breaking change next major version. Not the prettiest but the least disruptive.

```suggestion
        [Obsolete("Type will change in next major version to long, meanwhile use LongId to ensure proper value returned.")]
        public int Id => GetEnvironmentInteger("SYSTEM_PULLREQUEST_PULLREQUESTID");
        
        /// <summary>
        /// Gets the ID of the pull request that caused this build.
        /// This value is set only if the build ran because of a Git PR affected by a branch policy.
        /// </summary>
        /// <value>
        ///   The ID of the pull request that caused this build.
        /// </value>
        public long LongId => GetEnvironmentLongInteger("SYSTEM_PULLREQUEST_PULLREQUESTID");
```
and 
``` diff
-        public bool IsPullRequest => Id > 0;
+         public bool IsPullRequest => LongId > 0;
```

Reply:
Good point, I didn't think of that possibility.

I've adapted the code to match your suggestion